upb/table.c

```diff
@@ -194,14 +194,15 @@ static bool lookup(const upb_table *t, lookupkey_t key, upb_value *v,
   }
 }
 
-/* The given key must not already exist in the table. */
-static void insert(upb_table *t, lookupkey_t key, upb_tabkey tabkey,
+static bool insert(upb_table *t, lookupkey_t key, upb_tabkey tabkey,
                    upb_value val, uint32_t hash,
-                   hashfunc_t *hashfunc, eqlfunc_t *eql) {
+                   hashfunc_t *hashfunc, eqlfunc_t *eql, bool allow_failure) {
```

Comment:
instead of an allow_failure bool, you should make this method be an unconditional try_insert and then have callers assert whichever way they want